### PR TITLE
Factor out OCI get_subnet_id

### PR DESCRIPTION
**Pull request by aciba90**
topic: Factor out OCI get_subnet_id

### CODE REVIEW

**Comment by aciba90** [comment]
Additional context:

https://github.com/canonical/pycloudlib/pull/203#pullrequestreview-998844305

File: pycloudlib/oci/utils.py [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # This file is part of pycloudlib. See LICENSE file for license information.
 """Utilities for OCI images and instances."""
 import time
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
```
[the remark this answers is not in the document]
Comment (aciba90) [reply pycloudlib/oci/utils.py]: `pylint` does not like the fact that `oci` sub-packages are [dynamically loaded](https://github.com/oracle/oci-python-sdk/blob/master/src/oci/__init__.py#L19).

I have disabled the lint error and improved it. Thanks for the comment.

**Review by TheRealFalcon — APPROVED** [review]
LGTM, nice refactor

**Comment by TheRealFalcon** [comment]
Whoops...maybe we should have kept the comment :smile: 

https://app.travis-ci.com/github/canonical/pycloudlib/jobs/574971181

**Pull request merged**